src/textual/drivers/linux_driver.py

```diff
@@ -23,6 +24,25 @@
     from ..app import App
 
 
+class ExceptionHandlingThread(Thread):
```

Comment:
Extending Thread doesn't seem warranted here. Could we not  wrap `self.run_input_thread` with a try except?

Reply:
That's what `ExceptionHandlingThread` is doing. You can't catch the exception normally because it is called by `Thread.start()`.

Or do you mean we could wrap the body of `run_input_thread()` in `try ... except`? That would work, but it makes `run_input_thread()` more complex than it needs to be and do things (terminating the application) it shouldn't care about. It also means that, if anyone ever would add code above the `try ... except`, any exception raised by that new code would be ignored. I think separating the functionality of the input thread and the termination of the application if that input thread raises makes maintenance much easier.

We could also pass a new method as the `Thread` target (e.g. `_handle_run_input_thread()`) that only catches exceptions from `run_input_thread()` and terminates the application in that case. That would make the separation of functionality more clear without an additional class.

We could also set [threading.exceptionhook](https://docs.python.org/3/library/threading.html#threading.excepthook) to a function that terminates the application. But that would affect all threads in the entire application. I don't think that's a good idea because developers might want to set that themselves. Breaking textual by configuring exception in your application would a pretty annoying bug. We would also have to bump the minimum Python version to 3.8.

The benefit of `ExceptionHandlingThread` is that it could be used by all textual threads to always terminate gracefully without affecting application developers' exception handling. I didn't look into using `ExceptionHandlingThread` for other threads because textual's code base is a bit daunting and I'm not sure what I should consider.

If you *really* don't like the new class, I'd recommend the `_handle_run_input_thread()` wrapper method. The other solutions wouldn't really solve the issue for good.

Reply:
I ws proposing a wrapper. I would just call it `_run_input_thread`.